# SRNAEC
A Small-RNA sequencing reads restoration tool, which aims to detect errors and correct them as much as possible for all types of small RNA sequqencing data.  

Our SRNAEC is a user-friendly tool and suits for different research needs. The SRNAEC program is written in C++11 and tested on Red Hat Enterprise Linux Workstation release 7.7. It is availble under an open-source license.


## Download & Usage

Download

	git clone https://github.com/XuanrZhang/SRNAEC
	cd SRNAEC
	chmod +x SmallRNA.sh
	

	
Usage: ./SRNAEC.sh -f [Input_File] -h [threshold] -t [the number of threads] -o [Output_File] 

	Required OPTIONS:
	-f [File_Name]: smallRNA sequence fasta dataset（fasta files）

	Optional OPTIONS:
	-t [the number of threads]: default is 8;
	-h [threshold]: 5;
	-o [Ouput_FileName]: default is correct_read.fastq;
	
Examples: 

	# test using simulated datasets in github folders.
	./SRNAEC.sh -f ./seq.fasta -h 10  -t 20 -o correct.fasta
	

  
## Data format
Input: A clean miRNA read dataset in fastq format(After adapter cutting)

	- read.fa : store the whole read data needed to be corrected.
	
Output: A corrected read dataset 

	- correct_read.fa :store corrected read data.
	
## Data materials

**Simulated data**：generated by scprits （gene_simu_mixerr.c / gene_simu_sub.c）

**Real Wet-lab data**: can be downloaded at the link (https://www.ncbi.nlm.nih.gov//bioproject/PRJNA202973), with the Accession Number PRJNA202973.

We also provide copies in SRNAEC/data file folder.

	
## Citation
Please cite the work "Small RNA sequencing reads restoration".

## Question
If any bugs during your running, please email to xuan.zhang-5@student.uts.edu.au
